### ISSUE TITLE
Integer sliders breaks when using a hex format string

### ISSUE THREAD

**Issue by GuyCarver** [ocornut/imgui#5165]
Version: 18711
Branch: master

VisualStudio/WIndows 64 bit

Back-ends: imgui_impl_dx12.cpp + imgui_impl_win32.cpp
Operating System: Windows 10

I have a view of C64 memory view using 16 bit hex addresses
I attempted to add a vertical slider to adjust the memory view
Displaying the slider index in hex ("%08x" or "%04x") format causes the slider to not work after the index range goes beyond 10 decimal (0xa).

To fix the issue I set the flag ImGuiSliderFlags_NoRoundToFormat

```
// Repro code
ImGui::Begin("Example Bug");
static int int_value = 0;
ImGui::VSliderInt("##int", ImVec2(18, 160), &int_value, 0, 0x10, "%04x");
ImGui::End();
```

Thank you

**Comment by ocornut** [comment]
While tracking this I noticed that https://github.com/ocornut/imgui/commit/3e8087458d7232b522ce10583c44b62bd1729012 introduced unnecessary code.  Removed and fixed one part of the issue, using the drag/slider will now work.
(Also see which was a more complex solution for this #3133)

HOWEVER, inputing the hex value here won't. That's because we omit passing format to `DataTypeApplyFromText()` intentionally because scanf() uses precision to limit number of characters parsed, so parsing "1234" with "%03d" would read only "123", instead we use a common format for each data type, which in this instance makes us lose the hexadecimal input/output.
We should probably change DataTypeApplyFromText() to accept a format string and manually strip precision section out of it before calling sscanf.
